library/core/src/iter/traits/iterator.rs

```diff
@@ -457,8 +457,10 @@ pub trait Iterator {
     /// In other words, it zips two iterators together, into a single one.
     ///
     /// If either iterator returns [`None`], [`next`] from the zipped iterator
-    /// will return [`None`]. If the first iterator returns [`None`], `zip` will
-    /// short-circuit and `next` will not be called on the second iterator.
+    /// will return [`None`].
+    /// If the zipped iterator has no more elements to return then each further attempt to advance
+    /// it will advance the first iterator zero or one times and not call [`next`] on the second
+    /// iterator.
```

Comment:
> If the zipped iterator has no more elements to return[,] then each further attempt to advance it will […] not call [`next`] on the second iterator.

This is not correct. This only applies to the case where the _first_ iterator has no more elements. If the first iterator is non-empty but the second one is empty, of course [`next`] will (or at least can) be called on the second iterator.

[`next`]: https://doc.rust-lang.org/std/iter/trait.Iterator.html#tymethod.next

Reply:
Right, how about?

```rust
    /// If the zipped iterator has no more elements to return then each further attempt to advance
    /// it will first advance the first iterator at most one time and if it still yielded an item
    /// advance the second iterator at most one time.
```